src/main/java/com/redhat/cloud/notifications/models/EventType.java

```diff
@@ -102,6 +107,17 @@ public void setDescription(String description) {
         this.description = description;
     }
 
+    public UUID getApplicationId() {
+        if (applicationId == null && application != null) {
+            applicationId = application.getId();
+        }
+        return applicationId;
+    }
+
+    public void setApplicationId(UUID applicationId) {
+        this.applicationId = applicationId;
```

Comment:
Just to be safe, should we set this `applicationId` in `setApplication` too (to keep them sync'ed)? WDYT? or maybe move the `Schema` annotations to `getApplicationId` and remove the variable altogether?

Reply:
`setApplication` is actually a useless setter that could be removed. Hibernate uses the field access strategy because we have an `@Id` annotation at field level.

I had a low priority task on my todo list for a while: remove all useless getters and setters. There are a few of them in the Hibernate entities. I'll take care of that later.

Reply:
We still need the variable `applicationId` field for tests BTW, so `getApplicationId` and `setApplicationId` can't be removed.